app/views/layouts/_facebook_script.html.erb

```diff
@@ -1,21 +1,75 @@
 <!-- Facebook script -->
-<div id="fb-root"></div>
 <script>
-window.fbAsyncInit = function() {
-  FB.init({
-    appId      : '<%= ENV["FACEBOOK_APP_ID"] %>',
-    xfbml      : true,
-    version    : 'v2.8'
+var bindFacebookEvents, defer, dispatch, handleEvent, initializeFacebookSDK;
+var loadFacebookSDK, loaded, restoreFacebookRoot, saveFacebookRoot;
+
+({defer, dispatch} = Turbolinks);
+
+loaded = false;
+
+$(function() {
+  loadFacebookSDK();
+  if (!window.fbEventsBound) {
+    return bindFacebookEvents();
+  }
+});
+
+handleEvent = function(eventName, handler) {
+  return document.addEventListener(eventName, handler, false);
+};
+
+handleEvent("DOMContentLoaded", function() {
+  return defer(function() {
+    return loaded = true;
+  });
+});
+
+handleEvent("turbolinks:load", function() {
+  if (loaded) {
+    return dispatch("page:load");
+  }
+});
+
+bindFacebookEvents = function() {
+  $(document)
+    .on('turbolinks:request-start', saveFacebookRoot)
+    .on('turbolinks:load', restoreFacebookRoot)
+    .on('page:load', function() {
+      return typeof FB !== "undefined" && FB !== null ? FB.XFBML.parse() : void 0;
+  });
+  return this.fbEventsBound = true;
```

Comment:
Are you sure `this` here  will be `window`? Maybe you should call `window.fbEventsBound = true;`